cmd/nex/rootfs.go

```diff
@@ -1,3 +1,389 @@
+		return err
+	}
+
+	return nil
+}
+
+func (r *RootFS) Initialize() error {
+	repo, err := remote.NewRepository(r.ociRef.RawRef)
+	if err != nil {
+		return err
+	}
+	if r.NoTLS {
+		repo.PlainHTTP = true
+	}
+	r.ociRef.Repo = repo
+
+	repo.Client = &auth.Client{
+		Cache: auth.DefaultCache,
+		Credential: func(ctx context.Context, registry string) (auth.Credential, error) {
+			switch r.ociRef.Repo.Reference.Registry {
+			case "docker.io":
+				authGet, err := http.Get(fmt.Sprintf("https://auth.docker.io/token?service=registry.docker.io&scope=repository:%s:pull", repo.Reference.Repository))
+				if err != nil {
+					return auth.Credential{}, fmt.Errorf("failed to fetch auth token: %w", err)
+				}
+				defer authGet.Body.Close()
+				authBody := make(map[string]any)
+				if err := json.NewDecoder(authGet.Body).Decode(&authBody); err != nil {
+					return auth.Credential{}, fmt.Errorf("failed to decode auth response: %w", err)
+				}
+				token, ok := authBody["token"].(string)
+				if !ok {
+					return auth.Credential{}, errors.New("failed to parse token")
+				}
+				cred := auth.Credential{
+					AccessToken: token,
+				}
+				return cred, nil
+			default:
+				return auth.EmptyCredential, nil
+			}
+		},
+	}
+
+	r.ociRef.Descriptor, err = repo.Resolve(context.TODO(), repo.Reference.Reference)
+	if err != nil {
+		return err
+	}
+
+	manifestBytes, err := repo.Fetch(context.TODO(), r.ociRef.Descriptor)
+	if err != nil {
+		return err
+	}
+
+	// •	OCI Index (v1.Index): application/vnd.oci.image.index.v1+json
+	//   •	Represents a multi-platform image.
+	//   •	Contains a list of manifests, each with a Platform field indicating the OS and architecture.
+	// •	OCI Manifest (v1.Manifest): application/vnd.oci.image.manifest.v1+json
+	//   •	Describes an image for a specific platform.
+	//   •	Contains a list of layers and a config object.
+	switch r.ociRef.Descriptor.MediaType {
+	case v1.MediaTypeImageManifest:
+		var manifest v1.Manifest
+		indexDecoder := json.NewDecoder(manifestBytes)
+		if err := indexDecoder.Decode(&manifest); err != nil {
+			return err
+		}
+		r.ociRef.Manifest = &manifest
+	case v1.MediaTypeImageIndex:
+		var index v1.Index
+		manifestDecoder := json.NewDecoder(manifestBytes)
+		if err := manifestDecoder.Decode(&index); err != nil {
+			return err
+		}
+		for _, manifest := range index.Manifests {
+			if manifest.MediaType == v1.MediaTypeImageManifest && manifest.Platform.OS == r.Os && manifest.Platform.Architecture == r.Arch {
+				manifestBytes, err := repo.Fetch(context.TODO(), manifest)
+				if err != nil {
+					return err
+				}
+				var manifest v1.Manifest
+				indexDecoder := json.NewDecoder(manifestBytes)
+				if err := indexDecoder.Decode(&manifest); err != nil {
+					return err
+				}
+				r.ociRef.Manifest = &manifest
+				break
+			}
+		}
+	}
+
+	if r.ociRef.Manifest == nil {
+		return errors.New("no matching manifest found")
+	}
+
+	return nil
+}
+
+func (r *RootFS) Validate() error {
+	return nil
```

Comment:
How will validation work?

Reply:
Kong runs the Validate function on all commands automatically, so by satisfying the interface, we will get validation for free.  That said, there are no validations from rootfs command yet

Reply:
Just added them